Applications/MaterialAssist/Contracts/Boards/Interfaces/IMaterialAssistClientBoards.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByIds(IEnumerable<string> ids);
+
+        /// <summary>
+        /// Get board entities by board code.
+        /// </summary>
+        /// <param name="boardCode"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCode(string boardCode);
+
+        /// <summary>
+        /// Get board entities by board codes.
+        /// </summary>
+        /// <param name="boardCodes"></param>
+        /// <returns></returns>
+        Task<IEnumerable<BoardEntity>> GetBoardEntitiesByBoardCodes(IEnumerable<string> boardCodes);
+
+        /// <summary>
+        /// Get storage locations.
+        /// </summary>
+        /// <returns></returns>
+        Task<IEnumerable<StorageLocation>> GetStorageLocations();
+
+        #endregion Read
+
+        #region Create
+
+        /// <summary>
+        /// Create a board instance for a certain boardType.
+        /// </summary>
+        /// <param name="boardTypeCode"></param>
+        /// <param name="boardEntity"></param>
+        /// <returns></returns>
+        Task<BoardEntity> CreateBoardEntity(string boardTypeCode, BoardEntity boardEntity);
+
+        /// <summary>
+        /// Create a board type.
+        /// </summary>
+        /// <param name="boardType"></param>
+        /// <returns></returns>
+        Task CreateBoardType(BoardType boardType);
+
+        #endregion Create
+
+        #region Update
+
+        /// <summary>
+        /// Update board entity dimensions by its id (#).
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="length"></param>
+        /// <param name="width"></param>
+        /// <returns></returns>
+        Task UpdateBoardEntityDimensions(string id, double length, double width);
+
+        /// <summary>
+        /// Update board entity comments by its id (#).
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="comments"></param>
+        /// <returns></returns>
+        Task UpdateBoardEntityComment(string id, string comments);
+
+        /// <summary>
+        /// Store a board by its code (#).
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="length"></param>
+        /// <param name="width"></param>
+        /// <param name="storageLocation"></param>
+        /// <returns></returns>
+        Task StoreBoardEntity(string id, int length, int width, StorageLocation storageLocation);
+
+        /// <summary>
+        /// Remove all board entities by its code (#). This is available for all ManagementTypes.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="deleteBoardFromInventory"></param>
+        /// <returns></returns>
+        Task RemoveAllBoardEntities(string id, bool deleteBoardFromInventory = false);
```

Comment:
I have added 3 methods for the removal of a board, same for edgebands.
When removing an item which is of ManagementType GoodsInStock or Stack which has a quantity > 2 we offer multiple options for removing these items out of the storage. Please also check the suggested routes in the Client itself as I am not 100% sure on how to use new parameter to differntiate between these 3 options.

Reply:
So what is your question here?

Reply:
if we should differntiate by using a param or having 3 different routes for the 3 options

Reply:
But it is the same route just with this param "DeletionType" at the end to decide which one.
If we do it by param then we most likely have a huge validation in our controller to catch all possible cases
By splitting it we could reduce the complexity of the controller.